packages/back-end/src/events/handlers/utils.ts

```diff
@@ -131,44 +131,49 @@ export const filterEventForEnvironments = ({
   }
 };
 
+// Some of the feature keys that change affect all enabled environments
+export const RELEVANT_KEYS_FOR_ALL_ENVS: (keyof ApiFeature)[] = [
+  "archived",
+  "defaultValue",
+  "prerequisites",
+  "project",
+  "valueType",
+];
+
 export const filterFeatureUpdatedNotificationEventForEnvironments = ({
   featureEvent,
   environments,
 }: {
   featureEvent: FeatureUpdatedNotificationEvent;
   environments: string[];
 }): boolean => {
-  // if the environments are not specified, notify for all environments
-  if (environments.length === 0) {
-    return true;
-  }
-
   const { previous, current } = featureEvent.data;
-  if (previous.archived && current.archived) {
+  if (previous.archived === true && current.archived === true) {
     // Do not notify for archived features
     return false;
   }
-  // Manual environment filtering
-  const changedEnvironments = new Set<string>();
 
-  // Some of the feature keys that change affect all enabled environments
-  const relevantKeysForAllEnvs: (keyof ApiFeature)[] = [
-    "archived",
-    "defaultValue",
-    "prerequisites",
-    "project",
-    "valueType",
-  ];
-  if (relevantKeysForAllEnvs.some((k) => !isEqual(previous[k], current[k]))) {
+  // if the environments are not specified, notify for all environments
+  if (environments.length === 0) {
```

Comment:
As I said in the previous PR, I personally like to enforce strict boundaries with nullable values. Have the function declare its expected data types and make sure the callers do respect that.

Typescript is an incredible tool for that and will notify on all calling sites when an expected argument type has changed or is incorrect.

This way, the code is more readable: I don't have to wonder why or under what circumstances environment is null or undefined and what this is supposed to mean in the internal logic of the function.

Otherwise, pushing nullable values everywhere usually end-up having interfaces and arguments types that are mostly nullable, which is pretty confusing and irrelevant to the function handling them.